Samples/iOS-Swift/iOS-Swift/ViewControllers/NibViewController.swift

```diff
@@ -1,15 +1,59 @@
 import Foundation
+import Sentry
 import UIKit
 
 class NibViewController: UIViewController {
     
     @IBOutlet var button: UIButton!
+    var span: Span?
+    var spanObserver: SpanObserver?
     
     override func viewDidLoad() {
         super.viewDidLoad()
         button.backgroundColor = .black
         
         button.layer.cornerRadius = 10
         button.layer.masksToBounds = true
+        
+        if let span = SentrySDK.span, let rootSpan = span.rootSpan() {
+            self.span = span
+            spanObserver = SpanObserver(span: rootSpan)
+            spanObserver?.performOnFinish {
+                self.assertTransaction()
+            }
+        }
```

Comment:
`m`: I think this is duplicated across multiple ViewController now. Can we please refactor this?